types/evidence.go

```diff
@@ -768,6 +764,15 @@ type PhantomValidatorEvidence struct {
 
 var _ Evidence = &PhantomValidatorEvidence{}
 
+// NewPhantomValidatorEvidence creates a new instance of the respective evidence 
+func NewPhantomValidatorEvidence(vote *Vote, lastHeightValidatorWasInSet int64) *PhantomValidatorEvidence {
+	if vote == nil || lastHeightValidatorWasInSet <= 0 { return nil }
```

Comment:
why are u duplicating ValidateBasic logic here?

Reply:
I was just replicating what DuplicateVoteEvidence does - which is add null checks

Reply:
it hides errors. that's not a good style. if anything, the implementation should panic or return an error if someone calls a constructor w/ nil vote

Reply:
I'd remove the check

Reply:
okay. I also thought about having the option to return an error or even to validate basic when we create the evidence

Reply:
but I will remove the checks on all of them except the constructors for DuplicateVoteEvidence and PotentialAmnesiaEvidence which need to order the votes correctly